package-res/resources/web/pentaho/lang/events/DidChange.js

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "../Event",
+  "../mixins/mixinChangeset"
+], function(Event, mixinChangeset) {
+  "use strict";
+
+  /**
+   * @name DidChange
+   * @memberOf pentaho.lang.events
+   * @description This event is triggered when changing properties values
+   * with {@link pentaho.type.complex#set|complex.set} occurs without any failures.
+   *
+   * @extends pentaho.lang.Event
+   * @event "did:change"
+   */
+  return Event.extend("pentaho.lang.events.DidChange",
```

Comment:
so, you agree to move
pentaho/lang/events -> pentaho/type/events
pentaho/lang/mixins -> pentaho/type/mixins

awesome